cosi/threshold/mod.go

```diff
@@ -1,35 +1,60 @@
 package threshold
 
 import (
+	"sync/atomic"
+
+	"go.dedis.ch/dela"
 	"go.dedis.ch/dela/cosi"
+	"go.dedis.ch/dela/cosi/threshold/types"
 	"go.dedis.ch/dela/crypto"
 	"go.dedis.ch/dela/mino"
 	"golang.org/x/xerrors"
 )
 
-// Threshold is a function that returns the threshold to reach for a given n.
-type Threshold func(int) int
-
 func defaultThreshold(n int) int {
 	return n
 }
 
+// OneThreshold is a threshold function to allow one failure.
+func OneThreshold(n int) int {
+	if n <= 0 {
+		return 0
+	}
+
+	return n - 1
+}
+
+// ByzantineThreshold is a threshold function to allow a threshold number of
+// failures and comply to the Byzantine Fault Tolerance theorem.
+func ByzantineThreshold(n int) int {
+	if n <= 0 {
+		return 0
+	}
+
+	f := (n - 1) / 3
+
+	return n - f
+}
+
 // CoSi is an implementation of the cosi.CollectiveSigning interface that is
 // using streams to parallelize the work.
 type CoSi struct {
-	mino   mino.Mino
-	signer crypto.AggregateSigner
-
-	Threshold Threshold
+	mino      mino.Mino
+	signer    crypto.AggregateSigner
+	threshold atomic.Value
```

Comment:
I would  comment that this MUST contain only a type `cosi.Threshold` since you don't do the check in the actor.